README.md

```diff
@@ -1,121 +1,115 @@
+| [Citation](#referencing-tslearn) | A citation for tslearn for scholarly articles |
 
 # Installation
-
-## Using conda
-
-The easiest way to install `tslearn` is probably via `conda`:
-```bash
-conda install -c conda-forge tslearn
+There are different alternatives to install tslearn:
+* PyPi: `python -m pip install tslearn`
+* Conda: `conda install -c conda-forge tslearn`
+* Git: `python -m pip install https://github.com/rtavenar/tslearn/archive/master.zip`
+
+In order for the installation to be successful, the required dependencies must be installed. For a more detailed guide on how to install tslearn, please see the [Documentation](https://tslearn.readthedocs.io/en/latest/?badge=latest#installation)
+
+# Getting started
+
+## 1. Getting the data in the right format
+tslearn expects the data to be 3D. The three dimensions correspond to the number of time series, the number of measurements per time series and the number of dimensions respectively (`n_ts, max_sz, d`). In order to get the data in the right format, different solutions exist:
+* [You can use the utility functions such as `to_time_series_dataset`](https://tslearn.readthedocs.io/en/latest/gen_modules/tslearn.utils.html#module-tslearn.utils)
+* [You can convert from other popular time series toolkits in Python](https://tslearn.readthedocs.io/en/latest/integration_other_software.html)
+* [You can load any of the UCR datasets in the required format.](https://tslearn.readthedocs.io/en/latest/gen_modules/tslearn.datasets.html#module-tslearn.datasets)
+* [You can generate synthetic data using the `generators` module](https://tslearn.readthedocs.io/en/latest/gen_modules/tslearn.generators.html#module-tslearn.generators)
+
+It should further be noted that tslearn [supports variable-length timeseries](https://tslearn.readthedocs.io/en/latest/variablelength.html).
+
+```python3
+>>> from tslearn.utils import to_time_series_dataset
+>>> my_first_time_series = [1, 3, 4, 2]
+>>> my_second_time_series = [1, 2, 4, 2]
+>>> my_third_time_series = [1, 2, 4, 2, 2]
+>>> X = to_time_series_dataset([my_first_time_series,
+                                my_second_time_series,
+                                my_third_time_series])
+>>> y = [0, 1, 1]
 ```
 
-## Using PyPI
+## 2. Data preprocessing and transformations
+Optionally, tslearn has several utilities to preprocess the data. In order to facilitate the convergence of different algorithms, you can [normalize time series](https://tslearn.readthedocs.io/en/latest/gen_modules/tslearn.preprocessing.html#module-tslearn.preprocessing). Alternatively, in order to speed up training times, one can [resample](https://tslearn.readthedocs.io/en/latest/gen_modules/preprocessing/tslearn.preprocessing.TimeSeriesResampler.html#tslearn.preprocessing.TimeSeriesResampler) the data or apply a [piece-wise transformation](https://tslearn.readthedocs.io/en/latest/gen_modules/tslearn.piecewise.html#module-tslearn.piecewise), such as SAX.
```

Comment:
For some people, normalizing has a very precise sense (make the series unit norm). Maybe scaling is a more general term. (In this case, maybe the transformed dataset should be renamed `X_scaled`).